src/core/Akka/Actor/RepointableActorRef.cs

```diff
@@ -276,9 +276,12 @@ public void ReplaceWith(ICell cell)
                 {
                     DrainSysMsgQueue(cell);
 
-                    foreach (var envelope in _messageQueue)
+                    while (_messageQueue.Count > 0)
                     {
-                        cell.SendMessage(envelope.Sender, envelope.Message);
+                        // roughly equal to what "poll" does
```

Comment:
Leaving an implementation comment to note that what I provided below is functionally equivalent to http://www.tutorialspoint.com/java/util/linkedlist_poll.htm